src/Build/BackEnd/Components/RequestBuilder/RequestBuilder.cs

```diff
@@ -110,6 +111,16 @@ internal class RequestBuilder : IRequestBuilder, IRequestBuilderCallback, IBuild
         private bool _isZombie = false;
 
         /// <summary>
+        /// The dedicated scheduler object.
+        /// </summary>
+        private static TaskScheduler _dedicatedScheduler;
+
+        /// <summary>
+        /// Gets the dedicated scheduler.
+        /// </summary>
+        private TaskScheduler DedicatedScheduler => _dedicatedScheduler ?? (_dedicatedScheduler = new DedicatedThreadsTaskScheduler());
```

Comment:
`private static TaskScheduler _dedicatedScheduler = new DedicatedThreadsTaskScheduler();` and then the static property should just return the field. The CLR promises type creation is thread safe, so assignment expressions to static fields is thread safe. Therefore no locking needed.